lib/cartopy/io/shapereader.py

```diff
@@ -123,17 +123,19 @@ def __init__(self, geometry, attributes):
 
 class BasicReader:
     """
-    Provide an interface for accessing the contents of a shapefile.
+    Provide an interface for accessing the contents of a shapefile with the
+    Python Shapefile Library (PyShp). See the PyShp
+    `Readme <https://pypi.org/project/pyshp/>`_ for more information.
 
     The primary methods used on a BasicReader instance are
     :meth:`~cartopy.io.shapereader.BasicReader.records` and
     :meth:`~cartopy.io.shapereader.BasicReader.geometries`.
 
     """
 
-    def __init__(self, filename, bbox=None):
+    def __init__(self, filename, **kwargs):
         # Validate the filename/shapefile
-        self._reader = reader = shapefile.Reader(filename)
+        self._reader = reader = shapefile.Reader(filename, **kwargs)
```

Comment:
```suggestion
        self._reader = reader = shapefile.Reader(filename, bbox=bbox, **kwargs)
```

Reply:
Thanks! This led to finding that PyShp has supporting the bounding box filtering for a while now. Since BasicReader uses lazy loading, bbox moved from here to the records and geometries calls.